src/iterator.jl

```diff
@@ -51,15 +51,6 @@ end
 
 # For testing purposes:
 
-Base.isapprox(iter1::TestIterator, iter2::TestIterator) = false
-function Base.isapprox(
-    iter1::TestIterator{T1,IS,IE},
-    iter2::TestIterator{T2,IS,IE};
-    kwargs...,
-) where {T1,T2,IS,IE}
-    return isapprox(iter1.data, iter2.data; kwargs...)
-end
-
```

Comment:
Technically this is breaking, but I am pretty sure they are not used except in this package.
But I could keep them and move them to deprecated, do you think?